ClothesApp/Infrastructure/Repositories/BrandsRepository.cs

```diff
@@ -0,0 +1,64 @@
+using Application.Repositories;
+using Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repositories;
+
+public class BrandsRepository : IBrandsRepository
+{
+    private readonly ClothesAppContext _dbContext;
+
+    public BrandsRepository(ClothesAppContext context)
+    {
+        _dbContext = context;
+    }
+
+    public async Task<Brand> Insert(Brand brand)
+    {
+        await _dbContext.Brands.AddAsync(brand);
+        await _dbContext.SaveChangesAsync();
+
+        return brand;
+    }
+
+    public async Task<Brand> Update(Brand brand)
+    {
+        _dbContext.Update(brand);
+        await _dbContext.SaveChangesAsync();
+
+        return brand;
+    }
+    
+    public async Task<Brand> DeleteBrandById(long id)
+    {
+        var brand = await _dbContext.Brands.FindAsync(id);
```

Comment:
Delete should be very simple and there should be no double db operations, also you do the same check on service layer and throw exception if brand is not found